antarest/study/business/link_management.py

```diff
@@ -51,38 +57,62 @@ class LinkManager:
     def __init__(self, storage_service: StudyStorageService) -> None:
         self.storage_service = storage_service
 
-    def get_all_links(self, study: RawStudy, with_ui: bool = False) -> t.List[LinkInfoDTO]:
+    def get_all_links(self, study: RawStudy) -> t.List[LinkInfoDTOType]:
         file_study = self.storage_service.get_storage(study).get_raw(study)
-        result = []
+        result: t.List[LinkInfoDTOType] = []
+
         for area_id, area in file_study.config.areas.items():
-            links_config: t.Optional[t.Dict[str, t.Any]] = None
-            if with_ui:
-                links_config = file_study.tree.get(["input", "links", area_id, "properties"])
+            links_config = file_study.tree.get(["input", "links", area_id, "properties"])
+
             for link in area.links:
-                ui_info: t.Optional[LinkUIDTO] = None
-                if with_ui and links_config and link in links_config:
-                    ui_info = LinkUIDTO(
-                        color=f"{links_config[link].get('colorr', '163')},{links_config[link].get('colorg', '163')},{links_config[link].get('colorb', '163')}",
-                        width=links_config[link].get("link-width", 1),
-                        style=links_config[link].get("link-style", "plain"),
-                    )
-                result.append(LinkInfoDTO(area1=area_id, area2=link, ui=ui_info))
+                link_properties = links_config[link]
+
+                link_creation_data: t.Dict[str, t.Any] = {"area1": area_id, "area2": link}
+                link_creation_data.update(link_properties)
+
+                link_data: LinkInfoDTOType
+                if StudyVersion.parse(study.version) < STUDY_VERSION_8_2:
+                    link_data = LinkInfoDTOBase(**link_creation_data)
+                else:
+                    link_data = LinkInfoDTO820(**link_creation_data)
+
+                result.append(link_data)
 
         return result
 
-    def create_link(self, study: RawStudy, link_creation_info: LinkInfoDTO) -> LinkInfoDTO:
+    def create_link(self, study: RawStudy, link_creation_info: LinkInfoDTOType) -> LinkInfoDTOType:
+        if link_creation_info.area1 == link_creation_info.area2:
+            raise LinkValidationError(f"Cannot create link on same area: {link_creation_info.area1}")
+
+        if StudyVersion.parse(study.version) < STUDY_VERSION_8_2 and isinstance(link_creation_info, LinkInfoDTO820):
+            if link_creation_info.filter_synthesis is not None or link_creation_info.filter_year_by_year is not None:
```

Comment:
I think we should rely on pydantic here instead:
we parse the object depending on the version, pydantic knows what are the valid fields and which ones are not.